### ISSUE TITLE
Column with different breakpoint obscures preceding column

### ISSUE THREAD

**Issue by nlwillia** [twbs/bootstrap#17998]
This is an unrealistic example, but I'd like to understand what's going on and whether it's valid grid code or not.
````
<div class="container">
  <div class="row">
    <div class="col-xs-12">column1</div>
    <div class="col-md-12">column2</div>
  </div>
</div>
````
What I'm finding is that below the breakpoint, the second column computes its height differently and ends up covering the first.  This doesn't happen if the second column uses the same breakpoint or includes that breakpoint in its definition.  In this [demo](http://www.bootply.com/pDUt8AL9TT), notice how the first column disappears when the second column breaks.

**Comment by twbs-lmvtfy** [comment]
Hi @nlwillia!

You appear to have posted a live example (http://s.bootply.com/render/vuqbKWqrKV), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 39, column 9: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
You need to add a clearfix div as mentioned in http://getbootstrap.com/css/#grid-responsive-resets
Example: http://www.bootply.com/k0prQveHqe

**Comment by nlwillia** [comment]
Even though the columns are the same height and the first is exactly 12 columns?  I read that section of the docs, but it didn't seem to apply.

**Comment by cvrebert** [comment]
If you inspect the elements, you'll see that the 2nd element in your example is actually twice the height of the 1st element; of course, that's not obvious at first glance since the 2nd element is on top of the 1st element.
I admit this is a weird edge case. It would be good to clarify this situation in the docs, but I've no idea how to word it.
@mdo Any ideas?

**Comment by mdo** [comment]
No ideas and since it's so edge case, I'm inclined to punt on it :grin: given it's low priority to other v3 and v4 work.

**Comment by ryantheleach** [comment]
In case anyone is interested. it's because the column falling back to full width, isn't being floated.
http://stackoverflow.com/questions/42969468/bootstrap-large-column-overlay-medium/42990539#42990539
